src/main/java/repositories/CustomerRepository.java

```diff
@@ -0,0 +1,135 @@
+package repositories;
+
+import entities.Customer;
+import entities.Product;
+import scripts.database.ConnectionCloser;
+import scripts.database.DbConnection;
+import scripts.interfaces.CRUD;
+import scripts.mappers.DbCustomerMapper;
+import scripts.mappers.DbProductMapper;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+
+import static scripts.constants.CustomerQueriesConstants.*;
+
+public class CustomerRepository implements CRUD<Customer, Integer> {
+    private ConnectionCloser connectionCloser;
+    private DbConnection connection;
+
+    public CustomerRepository(DbConnection connection) {
+        this.connection = connection;
+
+        connectionCloser = new ConnectionCloser();
+    }
+
+    @Override
+    public void create(Customer customer) {
+        PreparedStatement pstmt = null;
+
+        try{
+            pstmt = connection.getOpenConnection().prepareStatement(INSERT_QUERY);
+            pullPreparedStatement(pstmt, customer);
+
+            int insert_rows = pstmt.executeUpdate();
+
+            System.out.println("CustomerRepository -> created "+insert_rows+" customer(s)");
+        } catch (SQLException ex){
+            throw new RuntimeException("Create Customer error", ex);
+        } finally {
+            connectionCloser.close(connection, pstmt);
+        }
+    }
+
+    @Override
+    public List<Customer> findAll() {
+        PreparedStatement pstmt = null;
+        ResultSet rs = null;
+
+        try {
+            pstmt = connection.getOpenConnection().prepareStatement(FIND_ALL_QUERY);
+            rs = pstmt.executeQuery();
+
+            List<Customer> customers = new DbCustomerMapper().mapAll(rs);
```

Comment:
Each time you call `findAll` method it will create new DbCustomerMapper object.

Add `DbCustomerMapper` as field and initialize in constructor 

```java
public class CustomerRepository ... {
   private DbCustomerMapper dbCustomerMapper;
   
   public CustomerRepository(...) {
       this.dbCustomerMapper = new ...;
   }

}

```
and use class field here.